src/components/ChannelItem.vue

```diff
@@ -1,37 +1,72 @@
+                <img loading="lazy" class="aspect-square w-[50%] rounded-full" :src="item.thumbnail" />
             </div>
             <p>
-                <span v-text="props.item.name" />
-                <font-awesome-icon v-if="props.item.verified" class="ml-1.5" icon="check" />
+                <span v-text="item.name" />
+                <font-awesome-icon v-if="item.verified" class="ml-1.5" icon="check" />
             </p>
         </router-link>
-        <p v-if="props.item.description" v-text="props.item.description" />
-        <router-link v-if="props.item.uploaderUrl" class="link" :to="props.item.uploaderUrl">
+        <p v-if="item.description" v-text="item.description" />
+        <router-link v-if="item.uploaderUrl" class="link" :to="item.uploaderUrl">
             <p>
-                <span v-text="props.item.uploader" />
-                <font-awesome-icon v-if="props.item.uploaderVerified" class="ml-1.5" icon="check" />
+                <span v-text="item.uploader" />
+                <font-awesome-icon v-if="item.uploaderVerified" class="ml-1.5" icon="check" />
             </p>
         </router-link>
 
-        <a v-if="props.item.uploaderName" class="link" v-text="props.item.uploaderName" />
-        <template v-if="props.item.videos >= 0">
-            <br v-if="props.item.uploaderName" />
-            <strong v-text="`${props.item.videos} ${$t('video.videos')}`" />
+        <a v-if="item.uploaderName" class="link" v-text="item.uploaderName" />
+        <template v-if="item.videos >= 0">
+            <br v-if="item.uploaderName" />
+            <strong v-text="`${item.videos} ${$t('video.videos')}`" />
         </template>
 
+        <button
+            v-if="subscribed != null"
+            class="btn w-max mt-2"
+            @click="subscribeHandler"
+            v-text="
+                $t('actions.' + (subscribed ? 'unsubscribe' : 'subscribe')) + ' - ' + numberFormat(item.subscribers)
+            "
+        />
+
         <br />
     </div>
 </template>
 
-<script setup>
-const props = defineProps({
-    item: {
-        type: Object,
-        required: true,
+<script>
+export default {
+    props: {
+        item: {
+            type: Object,
+            required: true,
+        },
+    },
+    data() {
+        return {
+            subscribed: null,
+        };
+    },
+    computed: {
+        channelId(_this) {
+            return _this.item.url.substr(-24);
+        },
+    },
+    mounted() {
+        this.updateSubscribedStatus();
+    },
+    methods: {
+        async updateSubscribedStatus() {
+            this.subscribed = await this.fetchSubscriptionStatus(this.channelId);
+            console.log(this.subscribed);
+        },
+        subscribeHandler() {
+            this.toggleSubscriptionState(this.channelId, this.subscribed).then(success => {
+                if (success) this.subscribed = !this.subscribed;
+            });
+        },
     },
-});
```

Comment:
We should not change this to the legacy Options API, we should be using the [Composition API](https://vuejs.org/guide/extras/composition-api-faq.html) like before ideally.

Ideally, we should migrate our existing code to the composition api.

Reply:
Looks awesome, quite similar to Svelte and a lot less boilerplate code :)

Reply:
How should we be using `this` now? The Vue docs say that we should migrate to replacing mixins with [Composable functions](https://vuejs.org/guide/reusability/composables), but that's certainly out of scope for this PR.

Reply:
Continuing the use of the composition API would require major refactors, which I don't intend to do with this PR.